src/simulation/elements/SLCN.cpp

```diff
@@ -0,0 +1,236 @@
+static int SLCN_SPARKLE[16] = {
+	PIXPACK(0x5a6679),
+	PIXPACK(0x6878a1),
+	PIXPACK(0xabbfdd),
+	PIXPACK(0x838490),
+	
+	PIXPACK(0xbccddf),
+	PIXPACK(0x82a0d2),
+	PIXPACK(0x5b6680),
+	PIXPACK(0x232c3b),
+
+	PIXPACK(0x485067),
+	PIXPACK(0x8b9ab6),
+	PIXPACK(0xadb1c1),
+	PIXPACK(0xc3c6d1),
+	
+	PIXPACK(0x8594ad),
+	PIXPACK(0x262f47),
+	PIXPACK(0xa9aebc),
+	PIXPACK(0xc2e1f7),
+};
+
+
+static int SLCN_SPARKLE_FX[16] = {
+	0,
+	0,
+	PMODE_FLARE | PMODE_GLOW,
+	0,
+
+	0,
+	0,
+	0,
+	0,
+
+	0,
+	0,
+	0,
+	0,
+
+	0,
+	0,
+	0,
+	PMODE_SPARK,
+};
+
+static const float SPARKLE_RATE = 1.0f;
+static const float VELOCITY_MULTIPLIER = 1.0f;
+static const float PHASE_THRESHOLD = 60.0f;
+
+static float clamp(float n, float low, float high) {
+  return std::max(low, std::min(n, high));
+}
+
+static int update(UPDATE_FUNC_ARGS)
+{
+	Particle &self = parts[i];
+
+	float velocity = std::abs(std::sqrt(std::pow(self.vx, 2) + std::pow(self.vy, 2)));
+
+	//FROM: GOLD.cpp:74 but prettied up a bit.
+	if (self.life == 0 && self.temp < 373.15f)
+	{
+		for (int j = 0; j < 4; j++)
+		{ 
+			int check_coords_x[] = { -4, 4, 0, 0 };
+			int check_coords_y[] = { 0, 0, -4, 4 };
+
+			int neighbour_x = check_coords_x[j];
+			int neighbour_y = check_coords_y[j];
+
+			int neighbour_id = pmap[y + neighbour_y][x + neighbour_x];
+
+			if (neighbour_id == 0) continue;
+
+			int neighbour_type = TYP(neighbour_id);
+			Particle &neighbour = parts[ID(neighbour_id)];
+
+			if (neighbour_type == PT_SPRK && neighbour.life != 0 && neighbour.life < 4)
+			{
+				sim->part_change_type(i, x, y, PT_SPRK);
+				self.life = 4;
+				self.ctype = PT_SLCN;
+			}
+		}
+	}
+
+	int gt_phase_count = 0;
+	
+	for (int neighbour_x=-2; neighbour_x<3; neighbour_x++)
+		for (int neighbour_y=-2; neighbour_y<3; neighbour_y++) 
+		{
+			int neighbour_id = pmap[y + neighbour_y][x + neighbour_x];
+
+			if (neighbour_id == 0) continue;
+
+			int neighbour_type = TYP(neighbour_id);
+			Particle &neighbour = parts[ID(neighbour_id)];
+			if (neighbour_type == PT_SLCN)
+			{
+				if (self.tmp == neighbour.tmp)
+				{
+					gt_phase_count += 1;
+				}
+			}
+		}
+
+	// Phase change logic.
+	int current_color_phase = self.tmp & 31; // limited to 0 through 31. using bitlogic for simplicity.
+	int current_spark_phase = self.tmp & 15; // limited to 0 through 15.
+
+	float &clr_phse_transition_cnt = self.pavg[0];
+	float &sprk_phse_transition_cnt = self.pavg[1];
+
+	// prevent these values from going through the roof.
+	clr_phse_transition_cnt = clamp(clr_phse_transition_cnt, 0.0f, PHASE_THRESHOLD * 2);
+	sprk_phse_transition_cnt = clamp(sprk_phse_transition_cnt, 0.0f, PHASE_THRESHOLD * 2);
+
+	clr_phse_transition_cnt += (velocity * VELOCITY_MULTIPLIER) + SPARKLE_RATE + std::abs(RNG::Ref().uniform01() * SPARKLE_RATE);
+	sprk_phse_transition_cnt += (velocity * VELOCITY_MULTIPLIER) + SPARKLE_RATE + std::abs(RNG::Ref().uniform01() * SPARKLE_RATE);
+
+	// evil trick to combat syncronization.
```

Comment:
I'd much rather you fixed the problems causing it, rather than sweeping it under the rug.

Reply:
This really is more part of the element's graphics code than anything else. It makes the RNG "more random" by smoothing out cases where splotches might start to appear. Also cleans up the output when SLCN is placed with a brush (again, removes splotchiness).